config/config.go

```diff
@@ -432,6 +432,18 @@ func (s *SourceSubnet) Validate(destinationBlockchainIDs *set.Set[string]) error
 
 	// Validate and store the allowed destinations for future use
 	s.supportedDestinations = set.Set[ids.ID]{}
+
+	// If the list of supported destinations is empty, populate with all of the configured destinations
+	if len(s.SupportedDestinations) == 0 {
+		for _, blockchainIDStr := range destinationBlockchainIDs.List() {
+			blockchainID, err := ids.FromString(blockchainIDStr)
+			if err != nil {
+				return fmt.Errorf("invalid blockchainID in configuration. error: %w", err)
+			}
+			s.supportedDestinations.Add(blockchainID)
+		}
+	}
```

Comment:
nit: can we return `nil` here to make it clear the below code doesn't apply

Reply:
Sure

Reply:
Not to nit pick a nit, but want to note that it's actually more error prone to return early here because it's likely for more validation (possibly not related to the supported destinations) to be added in the future, and if we didn't remove the early return, future validation would not be hit.

I think generally, it's good practice for these types of methods to only have a single "successful" return case at the end, and all early returns be known error cases.

Reply:
Yeah, I'm not too fussed with however we implement this, but I think my ideal solution would be to wrap the for loop starting on line 448 in an `else` clause to this `if`. I normally hate unnecessary nesting, but right now it's not clear from a first pass that these two loops are related.

Reply:
Ah that's a good point. I've reverted this change.